java8/src/main/java/fj/data/Java8.java

```diff
@@ -28,95 +29,95 @@ private Java8() {
-        return Java8.<A, B>F_Function().f(f);
+        return f::f;
     }
 
     public static <A, B> F<F<A, B>, Function<A, B>> F_Function() {
-        return f -> f::f;
+        return Java8::F_Function;
     }
 
     public static <A, B, C> F2<A, B, C> BiFunction_F2(final BiFunction<A, B, C> f) {
-        return Java8.<A, B, C>BiFunction_F2().f(f);
+        return f::apply;
     }
 
     public static <A, B, C> F<BiFunction<A, B, C>, F2<A, B, C>> BiFunction_F2() {
-        return f -> f::apply;
+        return Java8::BiFunction_F2;
     }
 
     public static <A, B, C> BiFunction<A, B, C> F2_BiFunction(final F2<A, B, C> f) {
-        return Java8.<A, B, C>F2_BiFunction().f(f);
+        return f::f;
     }
 
     public static <A, B, C> F<F2<A, B, C>, BiFunction<A, B, C>> F2_BiFunction() {
-        return f -> f::f;
+        return Java8::F2_BiFunction;
     }
 
     public static <A, E extends Exception> Supplier<Validation<E, A>> TryCatch0_Supplier(final Try0<A, E> t) {
-        return Java8.<A, E>TryCatch0_Supplier().f(t);
+        return () -> Try.f(t)._1();
     }
 
     public static <A, E extends Exception> F<Try0<A, E>, Supplier<Validation<E, A>>> TryCatch0_Supplier() {
-        return t -> () -> Try.f(t)._1();
+        return Java8::TryCatch0_Supplier;
     }
 
     public static <A, B, E extends Exception> Function<A, Validation<E, B>> TryCatch1_Function(final Try1<A, B, E> t) {
-        return Java8.<A, B, E>TryCatch1_Function().f(t);
+        return a -> Try.f(t).f(a);
     }
 
     public static <A, B, E extends Exception> F<Try1<A, B, E>, Function<A, Validation<E, B>>> TryCatch1_Function() {
-        return t -> a -> Try.f(t).f(a);
+        return Java8::TryCatch1_Function;
     }
 
     public static <A, B, C, E extends Exception> BiFunction<A, B, Validation<E, C>> TryCatch2_BiFunction(final Try2<A, B, C, E> t) {
-        return Java8.<A, B, C, E>TryCatch2_BiFunction().f(t);
+        return (a, b) -> Try.f(t).f(a, b);
     }
 
     public static <A, B, C, E extends Exception> F<Try2<A, B, C, E>, BiFunction<A, B, Validation<E, C>>> TryCatch2_BiFunction() {
-        return t -> (a, b) -> Try.f(t).f(a, b);
+        return Java8::TryCatch2_BiFunction;
     }
 
     public static <A> java.util.stream.Stream<A> List_JavaStream(final List<A> list) {
         return Iterable_JavaStream(list);
     }
 
     public static <A> Option<A> Optional_Option(final Optional<A> o) {
-        return Java8.<A>Optional_Option().f(o);
+        return o.isPresent() ? Option.some(o.get()) : Option.none();
     }
 
     public static <A> F<Optional<A>, Option<A>> Optional_Option() {
-        return o -> o.isPresent() ? Option.some(o.get()) : Option.none();
+        return Java8::Optional_Option;
     }
 
     public static <A> Optional<A> Option_Optional(final Option<A> o) {
-        return Java8.<A>Option_Optional().f(o);
+        return o.option(Optional.empty(), Optional::ofNullable);
```

Comment:
Should we really map `Some(null)` to `Optional.absent()`?
It is weird as the reverse mapping of `Optional.absent()` is `None()`.
@functionaljava can I change that and throw an `IllegalArgumentException` instead?
This would allow a lawful `Iso`.